grails-app/controllers/org/pih/warehouse/api/LocationApiController.groovy

```diff
@@ -242,7 +242,12 @@ class LocationApiController extends BaseDomainApiController {
         if (existingLocation.isZoneLocation() && Location.findAllByZone(existingLocation)) {
             throw new IllegalArgumentException("You cannot delete zone that is assigned to a bin location ${params.id}")
         }
-        locationService.deleteLocation(existingLocation)
+
+        try {
+            locationService.deleteLocation(existingLocation)
+        } catch (Exception e) {
```

Comment:
It would be best to avoid catching exceptions in API controllers and let generic error handlers deal with the exception.

My first suggestion was to maybe find the code for this exception and add it to `messages.properties` file, but I think this only works for validation errors and this is a database constraint violation which probably doesn't have an error code.

Could you look into how we can handle these kinds of errors generically in the ErrorsControler if it is possible, and if not I think try/catch here should be fine, but please change the Exception you are catching to `org.springframework.dao.DataIntegrityViolationException e` so that we let through other exceptions and let the the ErrorsController handle them, unless you dissagree.